scap-driver: fails to compile with kernel 6.4

A new kernel release is coming soon, hooray! The driver no longer builds, oh noes!

As reported in https://bugs.gentoo.org/906709:
```
  CC [M]  /var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/ppm_cputime.o
In file included from ./include/linux/linkage.h:7,
                 from ./arch/x86/include/asm/cache.h:5,
                 from ./include/linux/cache.h:6,
                 from ./arch/x86/include/asm/current.h:9,
                 from ./include/linux/sched.h:12,
                 from ./arch/x86/include/asm/syscall.h:14,
                 from /var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c:19:
/var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c: In function 'scap_init':
./include/linux/export.h:27:22: error: passing argument 1 of 'class_create' from incompatible pointer type [-Werror=incompatible-pointer-types]
   27 | #define THIS_MODULE (&__this_module)
      |                     ~^~~~~~~~~~~~~~~
      |                      |
      |                      struct module *
/var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c:2497:36: note: in expansion of macro 'THIS_MODULE'
 2497 |         g_ppm_class = class_create(THIS_MODULE, DRIVER_DEVICE_NAME);
      |                                    ^~~~~~~~~~~
In file included from ./include/linux/device.h:31,
                 from ./include/linux/cdev.h:8,
                 from /var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c:26:
./include/linux/device/class.h:229:54: note: expected 'const char *' but argument is of type 'struct module *'
  229 | struct class * __must_check class_create(const char *name);
      |                                          ~~~~~~~~~~~~^~~~
/var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.c:2497:23: error: too many arguments to function 'class_create'
 2497 |         g_ppm_class = class_create(THIS_MODULE, DRIVER_DEVICE_NAME);
      |                       ^~~~~~~~~~~~
./include/linux/device/class.h:229:29: note: declared here
  229 | struct class * __must_check class_create(const char *name);
      |                             ^~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:252: /var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src/main.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:2026: /var/tmp/portage/dev-util/scap-driver-0.29.3-r2/work/libs-e5c53d648f3c4694385bbe488e7d47eaa36c229a_build/driver/src] Error 2
make: *** [Makefile:16: all] Error 2
```
Introduced by mainline kernel commit [1aaba11da9aa](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=1aaba11da9aa7d7d6b52a74d45b31cac118295a1) which removed one of the arguments to `class_create`.

--- a/driver/main.c
+++ b/driver/main.c
@@ -2912,7 +2912,11 @@ int scap_init(void)
 		goto init_module_err;
 	}
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
 	g_ppm_class = class_create(THIS_MODULE, DRIVER_DEVICE_NAME);
+#else
+	g_ppm_class = class_create(DRIVER_DEVICE_NAME);
+#endif
 	if (IS_ERR(g_ppm_class)) {
 		pr_err("can't allocate device class\n");
 		ret = -EFAULT;

